test/starlark_tests/rules/generate_xcframework.bzl

```diff
@@ -134,6 +137,34 @@ def _create_xcframework(
         ])
         args.extend(["-framework", framework_path])
 
+        if dsyms and framework_path in dsyms:
+            dsym_files = dsyms[framework_path]
+
+            inputs.extend(dsym_files)
+
+            dsym_bundles_set = sets.make()
+
+            for file in dsym_files:
+                bundle_extension = ".framework.dSYM"
+                prefix, ext, _ = file.short_path.partition(bundle_extension)
+                bundle_path = prefix + ext
```

Comment:
lifted this out of `bundle_paths.bzl` to appease buildifier visibility conventions